File: projects/Project4/factorial.rb
#! /usr/bin/ruby
# factorial.rb calculates the factorial of an integer given from the user
# Created by: Ethan Clark for CS214 Project4 at Calvin College
# Date: March 2, 2016
#########################################################################

# Input: an integer value
# Precondition: the integer is positive
# Output: the corresponding factorial

def factorial value
	
	# Intialize the return value to be 1
	answer = 1

	#For loop to calculate the factorial of value
	for i in 2..value
		answer = answer * i
	end

	# Return the factorial to the main program
	return answer
end

# If this is the main file running
if __FILE__ == $0
	
	# Prompt the user to enter an integer
	print "Enter an integer: "
	value = gets.chomp.to_i

	# Calculate the factorial
	new_value = factorial(value)

	# Print out the factorial to the user
	print "\nThe factorial of #{value} is #{new_value}\n"

end

File: projects/Project4/Factorial.java
/* Factorial.java computes the factorial of a user input integer
 *
 * Input: a positive integer, n
 * Precondition: a positive integer
 * Output: the factorial (!) of the integer
 *
 * Created by: Ethan Clark for CS 214 Homework4 at Calvin College
 * Date: February 29, 2016
 ********************************************************************/

import java.util.Scanner;

public class Factorial
{
	public static void main(String[] args) {
		Scanner keyboard = new Scanner(System.in);

		// Prompt the user to enter an integer
		System.out.println("\nCalculating factorial...");
		System.out.println("\nPlease enter an integer: ");

		// Read in the string, convert it to an int
		String word = keyboard.next();
		int number = Integer.parseInt(word);

		// Output the factorial to the user
		System.out.println("The factorial of " + number + " is " + factorial_calc(number));
	}

	/* function factorial_calc() computes the factorial, given the integer
	 * Parameter: int number
	 * Precondition: number is a positive integer
	 * Returns: the calculated factorial
	 */
	public static int factorial_calc(int number) {

		int answer = 1;				// Initialize the return answer to be 1

		for (int i = 2; i <= number; i++) {	// For loop to multiply every number from 2
			answer *= i;				// to number to the answer variable
		}

		return answer;				// Return answer
	}
}
